src/twisted/plugins/cred_unix.py

```diff
@@ -17,22 +17,23 @@
 from twisted.cred.credentials import IUsernamePassword
 from twisted.cred.error import UnauthorizedLogin
 from twisted.internet import defer
+from twisted.python.compat import StringType
 
 
 
 def verifyCryptedPassword(crypted, pw):
-    if crypted[0] == '$': # md5_crypt encrypted
```

Comment:
Is there any chance this breaks MD5  encrypted passwords on python 2?  It looks like the parsing here relates to a GLibc implemenation detail (http://man7.org/linux/man-pages/man3/crypt.3.html), and the crypt module in p2 doesn't have those methods.  A secondary question is of course does it matter (security vs compatibility argument)?  I'm not overly familiar with the crypt API, so I don't have test to add right this second, but it might be worth trying.

Reply:
On my Python 3 system, I ran this program to generate an MD5 password:

```
import crypt
crypt.crypt("password", crypt.METHOD_MD5)
```

I took the string which was generated: `'$1d2n7Q0.r54s'`

and ran the following program in a Python 2.6.9 virtualenv:

```
import crypt
crypted =  "$1d2n7Q0.r54s"
crypt.crypt("password", "$1d2n7Q0.r54s") == crypted
```

and the Python 2.6.9 program returned `True`.  So this doesn't break
parsing of MD5 passwords.

Reply:
Other thing to note, is that in the Python 2 documentation:
https://docs.python.org/2/library/crypt.html

there is this:

```
Since a few crypt(3) extensions allow different values, with different sizes in the salt, it is recommended to use the full crypted password as salt when checking for a password.
```

Reply:
Also, to confirm with the implementation, this is where crypt is called in CPython 2.7:

https://github.com/python/cpython/blob/2.7/Modules/cryptmodule.c#L27

and this the implementation of crypt inside glibc:

https://sourceware.org/git/?p=glibc.git;a=blob;f=crypt/crypt-entry.c;h=2271665af2d21d2aec61ebe712ffc946c035e61e;hb=HEAD#l74

If you notice in the glibc code, the crypt() implementation parses the first
few characters of the salt, and depending on what prefix is there,
it hands it off to the code which parses MD5, SHA512, SHA256, etc.

So the change in this PR is compatible with the underlying implementation.